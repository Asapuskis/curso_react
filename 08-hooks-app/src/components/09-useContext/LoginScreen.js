import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const LoginScreen = () => {

    const { setUser } = useContext(UserContext);

    return (
        <>
            <h1>LoginScreen</h1>
            <hr/>
            {/* AÃ±adimos un user para demostrar que podemos utilizar las funciones pasadas por context */}
            <button
                className='btn btn-primary'
                onClick={ () => setUser( {
                    id: 123,
                    name: 'Paco'
                })}>
                Login
            </button>
        </>
    )
}
