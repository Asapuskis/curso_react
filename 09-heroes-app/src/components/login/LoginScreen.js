import { useNavigate } from 'react-router-dom';

export const LoginScreen = () => {

  // Para hacer la navegación, podemos utilizar el hook useNavigate creado por react-router-dom, la propia librería de rutas.
  const navigate = useNavigate();

  const handleLogin = () => {
    console.log('handleLogin!');
    // La función navigate cambia la localización a la ruta indicada.
    // Si se quiere, se puede enviar el parámetro replace, que si está a true, elimina el historial de navegación dentro de la web
    // para evitar volver a zonas no deseadas.
    navigate('/marvel', {
      replace: true
    });
  }

  return (
    <div className='container mt-5'>
        <h1>LoginScreen</h1>
        <hr />
        <button
          className='btn btn-primary'
          onClick={ handleLogin }
        >
          Login
        </button>
    </div>
  )
}
