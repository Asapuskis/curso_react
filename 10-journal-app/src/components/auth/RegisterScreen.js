import React from 'react'
import { Link } from 'react-router-dom'


export const RegisterScreen = () => {
  return (
    <>
      <h3 className='auth__title'>Register</h3>
      <form>
        <input className='auth__input' autoComplete='off' type="text" placeholder="Name" name='name'/>
        <input className='auth__input' autoComplete='off' type="text" placeholder="Email" name='email'/>
        <input className='auth__input' type="password" placeholder="Password" name='password'/>
        <input className='auth__input' type="password" placeholder="Confirm Password" name='confirmpassword'/>
        <button className='btn btn-primary btn-block' type='submit'>Login</button>
        <Link className='link' to='/auth/register' >Already Registered?</Link>
      </form>
    </>
  )
}
