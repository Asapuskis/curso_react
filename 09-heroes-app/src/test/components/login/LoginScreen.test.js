import { mount } from "enzyme";
import { MemoryRouter, Routes, Route } from "react-router-dom";
import { AuthContext } from "../../../auth/authContext";
import { LoginScreen } from "../../../components/login/LoginScreen";


const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ( {
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockNavigate
}));

describe('Pruebas en LoginScreen', () => {

    const contextValue = {
        dispatch: jest.fn(),
        user: {
            logged: false
        }
    }

    const wrapper = mount(
        <AuthContext.Provider value={ contextValue }>
            <MemoryRouter initialEntries={ ['/login' ]}>
                <Routes>
                    <Route path="/login" element={ <LoginScreen /> } />
                </Routes>
            </MemoryRouter>
        </AuthContext.Provider>
    );

    test('Debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('debe realizar el dispatch y la navegaciÃ³n', () => {
        const handleClick = wrapper.find('button').prop('onClick');
        handleClick();

        expect(mockNavigate).toHaveBeenCalledWith('/marvel', { replace: true });
        localStorage.setItem('lastPath', '/dc');
        handleClick();

        expect(mockNavigate).toHaveBeenCalledWith('/marvel', { replace: true });
    });

});