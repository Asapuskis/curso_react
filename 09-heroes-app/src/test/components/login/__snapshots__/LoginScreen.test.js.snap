// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en LoginScreen Debe mostrarse correctamente 1`] = `
<div
  className="container mt-5"
>
  <h1>
    LoginScreen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
